mailgun/test_scripts/tlt-2306-empty-bodies-integration-test.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python
+
+import copy
+import hashlib
+import hmac
+import json
+import time
+import uuid
+
+import requests
+from django.conf import settings
+
+try:
+    settings.NO_REPLY_ADDRESS
+except ImportError:
+    # make sure the project root is in sys.path
+    import os, sys
+    sys.path.insert(0, os.path.normpath(
+                           os.path.join(os.path.abspath(__file__), '..', '..', '..')))
+
+def generate_signature_dict():
+    timestamp = str(time.time())
+    token = str(uuid.uuid4())
+    signature = hmac.new(
+        key=settings.LISTSERV_API_KEY,
+        msg='{}{}'.format(timestamp, token),
+        digestmod=hashlib.sha256
+    ).hexdigest()
+    return {
+        'timestamp': timestamp,
+        'token': token,
+        'signature': signature
+    }
+
+# NOTE: this list contains only test accounts, and should be configured as
+#       world sendable
+list_address = 'canvas-206-3824@mg.dev.tlt.harvard.edu'
+
+# prep the base post body, which has no body params at all
+base_post = {
+    'sender': 'Integration Test <integrationtest@example.edu>',
+    'from': settings.NO_REPLY_ADDRESS,
+    'recipient': list_address,
+    'subject': 'blah',
+    'To': list_address,
+}
+base_post.update(generate_signature_dict())
+
+# generate the 4 "empty body cases"
+bodies = [
+    {},
+    {'body-html': '', 'body-plain': ''},
+    {'body-html': ''},
+    {'body-plain': ''},
+]
+posts = []
+for body in bodies:
+    post = copy.deepcopy(base_post)
+    post.update(body)
+    posts.append(post)
```

Comment:
Would it be helpful to make the subject of the email a string repr of the `body`, e.g. 'testing case where body={...}' ?